### ISSUE TITLE
Data tab should reflect changes in the columns of the data file

### ISSUE THREAD

**Issue by torronen** [dotnet/machinelearning-modelbuilder#1604]
**System Information (please complete the following information):**
latest versions as of 14.7.2021, windows 10

**Describe the bug**
If the CSV file has changed, by adding new or removing old columns, it is not reflected in Data tab. If trying to retrain with a changed CSV file, it may fail because it tries to read wrong column, or worse, it may not fail but uses wrong columns from CSV to each feature.

**To Reproduce**
1. Train a model with any CSV file
2. Edit CSV file and add new columns (or edit .mbconfig JSON to point to a new CSV file)
3. Come back and observer columns shown in Data view. Note that the new columns are not visible.

**Expected behavior**
Data tab seems to re-read the file (because it is slow to open), it also should update the columns if any changes.

**Additional context**
Low priority. Workaround is to remove CSV file and then re-select it in the Data tab.

**Comment by beccamc** [comment]
@torronen Thanks for the feedback. When the user selects a file we process it's columns and predict their type. When re-loading an mbconfig file we do not re-process the .csv file because it would wipe out manual changes you made. As you predicted, it's low priority to optimize this case as you can just reload the .csv. 

Our team is working on how data set versioning should work. cc: @romalpani and @briacht. Do we have a model & data versioning issue tracking this?

**Comment by torronen** [comment]
Makes sense. What about taking the column index from the actual file, but ignore other changes? Now I am afraid it reads based on original col indices, but my team keeps adding new data as they are able to source it. Probably they should have versioning in file names. Anyway I am concerned I might not notice some change and model builder still runs because data type was same. For example, use newly added outside temperature instead of indoor temperature.


I was trying to signal this does not affect my work much, but it does a little bit. Would "low impact" be a good term for it, or how should I call it?

**Comment by torronen** [comment]
How about adding a "Validate data" button to check the current columns match the ones in .mbconfig? I encountered this issue when working with old .mbconfig files but I no longer have the exact original files and I am unsure if the updates ones have any changes in the columns or their order.

**Comment by beccamc** [comment]
Thanks for the additional ideas! Pulling in design, as this will be a longer term decision on how we should handle dataset changes over time.
